backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java

```diff
@@ -54,19 +61,133 @@ public Response setCommit(
     }
   }
 
+  @Override
+  public Response setCommitFromDatasetVersion(
+      DatasetVersion datasetVersion,
+      DatasetDAO datasetDAO,
+      BlobDAO blobDAO,
+      RepositoryDAO repositoryDAO,
+      MetadataDAO metadataDAO,
+      FileHasher fileHasher)
+      throws ModelDBException, NoSuchAlgorithmException {
+    try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+      session.beginTransaction();
+      DatasetBlob.Builder datasetBlobBuilder = DatasetBlob.newBuilder();
+      Blob.Builder blobBuilder = Blob.newBuilder();
+      blobBuilder.addAllAttributes(datasetVersion.getAttributesList());
+      switch (datasetVersion.getDatasetVersionInfoCase()) {
+        case RAW_DATASET_VERSION_INFO:
+        case QUERY_DATASET_VERSION_INFO:
+          throw new ModelDBException("Not supported", Code.UNIMPLEMENTED);
+        case PATH_DATASET_VERSION_INFO:
+          PathDatasetVersionInfo pathDatasetVersionInfo =
+              datasetVersion.getPathDatasetVersionInfo();
+          List<DatasetPartInfo> partInfos = pathDatasetVersionInfo.getDatasetPartInfosList();
+          Stream<PathDatasetComponentBlob> result = partInfos.stream().map(this::componentFromPart);
+          if (pathDatasetVersionInfo.getLocationType() == PathLocationType.S3_FILE_SYSTEM) {
+            datasetBlobBuilder.setS3(
+                S3DatasetBlob.newBuilder()
+                    .addAllComponents(
+                        result
+                            .map(path -> S3DatasetComponentBlob.newBuilder().setPath(path).build())
+                            .collect(Collectors.toList())));
+          } else {
+            datasetBlobBuilder.setPath(
+                PathDatasetBlob.newBuilder().addAllComponents(result.collect(Collectors.toList())));
+          }
+          break;
+        case DATASETVERSIONINFO_NOT_SET:
+          throw new ModelDBException("Wrong dataset version type", Code.INVALID_ARGUMENT);
+      }
+      List<BlobContainer> blobList =
+          Collections.singletonList(
+              BlobContainer.create(
+                  BlobExpanded.newBuilder()
+                      .setBlob(blobBuilder.setDataset(datasetBlobBuilder))
+                      .build()));
```

Comment:
where are we adding the location?

Reply:
I have added it in my follow up PR: https://github.com/VertaAI/modeldb/pull/752/files#diff-29aea182137a8206d4efee5dc0699386R103

Reply:
I think it's better to keep in Anand's PR because he fixed there some issues he found during implementing create dataset version and also did some refactoring